doc/examples/type_hints.rst

```diff
@@ -94,24 +94,33 @@ Typed Collection
 
 You can use :py:class:`~typing.TypedDict` (Python 3.8+) when using a well-defined schema for the data in a
 :class:`~pymongo.collection.Collection`. Note that all `schema validation`_ for inserts and updates is done on the server.
-These methods automatically add an "_id" field.
+These methods automatically add an "_id" field. In the example below the "_id" field is
+marked by the :py:class:`~typing_extensions.NotRequired` notation to allow it to be accessed when reading from
+``result``. If it is simply not included in the definition, then it will be automatically added, but it will raise a
+type-checking error if you attempt to access it. Another option would be to generate the "_id" field yourself, and make
+it a required field. This would give the expected behavior, but would then also prevent you from relying on PyMongo to
+insert the "_id" field.
```

Comment:
We should document that `typing_extensions` has to be installed for `NotRequired` unless using Python 3.11+.